### DISCUSSION TITLE
Xamarin to MAUI app migration problem with Job

### DISCUSSION THREAD

**Discussion by AngeloLazzari** [shinyorg/shiny#1531] [Q&A]
Good morning, first of all thank you for this library!!!!

I am migrating from Xamarin to MAUI and we already call the jobmanager with success for a sincro job we have.

Now we have this call

`Shiny.Hosting.Host.GetService<IJobManager>().RunTask("LoadMap", async (token) =>
{//do somthing}`

but this call runs into a null because we didn't set the 
`            builder.Services.AddJob(typeof(SincroJob),"Sincro");
`
as we did for the sincro job here...

the problem is that for this "LoadMap" job we don't have a class that implements the IJob interface, so I don't know how to instance the obj...

can you please help me? 

Alsa I notices that in your documentation I could use the IJobManager to call, delete or query a job state... but do I have to substitute the actual Shiny.Hosting.Host.GetService<IJobManager>().RunTask with, for example a var result = await manager.Run("YourJobName");?

Thank you very much!

**Comment by aritchie** [comment]
https://shinylib.net/client/appbuilder/
